spec/rubocop/cop/style/redundant_heredoc_delimiter_quotes_spec.rb

```diff
@@ -152,4 +152,13 @@
       EDGE　CASE
     RUBY
   end
+
+  it 'does not register an offense when using the heredoc delimiter with escapes' do
+    expect_no_offenses(<<~'RUBY')
+      do_something(<<~'EOS')
+        Preserve \
```

Comment:
This tests `\\` in `STRING_INTERPOLATION_OR_ESCAPED_CHARACTER_PATTERN`.

See https://github.com/rubocop/rubocop/pull/11528/files#r1097249283